### PR TITLE
PAPP-37955: fix: allow empty packaged dependency sets

### PR DESCRIPTION
**Pull request by sodle-splunk**
## Summary

Allow the static dependency validator to accept an empty generated `pip313_dependencies`
dictionary while continuing to require the key and dictionary shape.

## Why

The dependency packager deliberately excludes packages supplied by SOAR. Apps such as Mimecast
and Automox can therefore have a nonempty `requirements.txt` but no wheels that need bundling. The
packager correctly writes `{"wheel": []}`, but the following static-test hook rejects that valid
output, leaving both connector PRs blocked.

## Changes

- Remove the incorrect assumption that every requirements file must produce at least one bundled
  Python 3.13 wheel.
- Preserve validation that `pip313_dependencies` exists when `requirements.txt` exists.
- Preserve validation that the generated value is a dictionary.
- Add regression tests for a valid empty generated wheel set and a missing generated key.

## Validation

- `pre-commit run --all-files`
- Regression tests under Python 3.9 (CI version): 2 passed
- Regression tests under Python 3.13: 2 passed
- Ruff lint and formatting checks
- Python compilation

## Tracking

- [PAPP-37955](https://splunk.atlassian.net/browse/PAPP-37955)
- Blocks [Mimecast connector PR #26](https://github.com/splunk-soar-connectors/mimecast/pull/26)
- Also blocks [Automox connector PR #6](https://github.com/splunk-soar-connectors/automox/pull/6)

Written by Codex.

### CODE REVIEW

File: local_hooks/tests/test_static_tests.py [hunk 1]
```diff
@@ -67,8 +70,8 @@ def test_static_tests(app_dir: Path):
     ["tests/data/static_tests/sdk_app_dir"],
     indirect=["sdk_app_dir"],
 )
-@patch("local_hooks.helpers.load_sdk_apps_enviornment")
-@patch("local_hooks.helpers.generate_sdk_app_manifest")
+@patch("local_hooks.app_tests.utils.app_parser.load_sdk_apps_enviornment")
```
Comment (sodle-splunk) [inline local_hooks/tests/test_static_tests.py]: can we fix the misspelling of `environment` while we're changing things?

**Review by phantom-jacob — APPROVED** [review]
